Source/WebKit/UIProcess/Extensions/Cocoa/API/WebExtensionContextAPIStorageCocoa.mm

```diff
@@ -194,27 +194,22 @@
 
     auto *changedData = [NSMutableDictionary dictionary];
 
-    if (!newKeysAndValues) {
-        [oldKeysAndValues enumerateKeysAndObjectsUsingBlock:^(NSString *key, NSString *value, BOOL *) {
-            changedData[key] = @{ oldValueKey: parseJSON(value, { JSONOptions::FragmentsAllowed }) ?: NSNull.null };
-        }];
-    } else {
-        [newKeysAndValues enumerateKeysAndObjectsUsingBlock:^(NSString *key, NSString *value, BOOL *) {
-            if (NSString *oldValue = oldKeysAndValues[key]) {
-                if (![oldValue isEqualToString:value]) {
-                    changedData[key] = @{
-                        oldValueKey: parseJSON(oldValue, { JSONOptions::FragmentsAllowed }) ?: NSNull.null,
-                        newValueKey: parseJSON(value, { JSONOptions::FragmentsAllowed }) ?: NSNull.null,
-                    };
-                }
-
-                return;
-            }
+    // Process new or changed keys.
+    [newKeysAndValues enumerateKeysAndObjectsUsingBlock:^(NSString *key, NSString *newValue, BOOL *) {
+        NSString *oldValue = oldKeysAndValues[key];
+
+        if (!oldValue || ![oldValue isEqualToString:newValue]) {
+            id parsedNewValue = parseJSON(newValue, JSONOptions::FragmentsAllowed) ?: NSNull.null;
+            id parsedOldValue = oldValue ? parseJSON(oldValue, JSONOptions::FragmentsAllowed) ?: NSNull.null : nil;
```

Comment:
Multiple ternary operators make this line a little bit tough to follow, but I'm not sure how much cleaner it can be made.